### PR TITLE
Issue #551: Search suggestion autocomplete

### PR DESCRIPTION
**Pull request by pascalchevrel**
* adds a new jquery libray that allows autocompletion
* base.php now allows defining files outside of the js folder (if the path starts with a /)

### CODE REVIEW

File: app/views/templates/base.php [hunk 1]
```diff
@@ -121,11 +121,12 @@
...
   <script src="/assets/clipboard.js/clipboard.js-built.js?v=<?= VERSION ?>"></script>
-
-<?php foreach ($javascript_include as $js_file):?>
-  <script src="/js/<?= $js_file . $cache_bust ?>"></script>
-<?php endforeach?>
-
+<?php
+foreach ($javascript_include as $js_file) {
+    $prefix = ($js_file[0] == '/') ? '' : '/js/';
+    print '  <script src="' . $prefix . $js_file . $cache_bust . '"></script>' . "\n";
```
Comment (flodolo) [inline app/views/templates/base.php]: Should we just update the existing includes instead? There are only 5 or 6 of them

File: web/js/base.js [hunk 2]
```diff
@@ -62,6 +62,30 @@ $(document).ready(function() {
 
     // Focus on the main search field.
     $('#recherche').focus();
+
+    // Autocomplete search suggestions in main search box using our own API
+    $('#recherche').autocomplete({
+        serviceUrl: function (query){
+            return 'https://transvision.mozfr.org/api/v1/tm/'
```
Comment (flodolo) [inline web/js/base.js]: We should query the local installation, not the remove prod server.

File: web/js/base.js [hunk 3]
```diff
@@ -62,6 +62,30 @@ $(document).ready(function() {
...
+                + $('#repository').val() + '/'
+                + $('#source_locale').val() + '/'
+                + $('#target_locale').val() + '/'
+                + query + '/';
+        },
+        params: 'max_results=10',
+        minChars: 2,
+        transformResult: function(response) {
+            data = JSON.parse(response);
```
Comment (flodolo) [inline web/js/base.js]:
```
var data = JSON.parse(response);
```

File: web/js/base.js [hunk 4]
```diff
@@ -62,6 +62,30 @@ $(document).ready(function() {
...
+            return {
+                suggestions: $.map(data, function(dataItem) {
+                    return { value: dataItem.source, data: dataItem.target };
```
Comment (flodolo) [inline web/js/base.js]: A find this style a bit clearer 
```JS
return { 
    value: dataItem.source, 
    data: dataItem.target 
};
```

File: web/js/base.js [hunk 5]
```diff
@@ -62,6 +62,30 @@ $(document).ready(function() {
...
+                })
+            };
+        },
+        onSelect : function() {
```
Comment (flodolo) [inline web/js/base.js]: Nit: no space before colon

File: web/style/transvision.css [hunk 6]
```diff
@@ -578,6 +578,37 @@ fieldset {
     width: 150px;
 }
 
+/* Autocomplete styles for suggestion popup */
+.autocomplete-suggestions {
+    border: 1px solid #999;
+    background: #FFF;
```
Comment (flodolo) [inline web/style/transvision.css]: HEX colors should be all lowercase for consistency, also `background-color` here is enough

File: web/style/transvision.css [hunk 7]
```diff
@@ -578,6 +578,37 @@ fieldset {
...
+}
+
+.autocomplete-suggestion {
+    padding: 2px 5px;
+    white-space: nowrap;
+    overflow: hidden;
+}
+
+.autocomplete-selected {
+    background: #F0F0F0;
```
Comment (flodolo) [inline web/style/transvision.css]: Same here

File: web/style/transvision.css [hunk 8]
```diff
@@ -578,6 +578,37 @@ fieldset {
...
+}
+
+.autocomplete-suggestions strong {
+    font-weight: normal;
+    color: #3399FF;
```
Comment (flodolo) [inline web/style/transvision.css]: Same here (lowercase)

**Comment by flodolo** [comment]
I haven't tested the code, but I think we should have the new JavaScript code inside of `main_search.js`, not `base.js`.

Besides a few nits, this works really well, but I find confusing that I only get suggestions from English. Not sure how much can be done to add a second API query and show both results.

File: app/views/templates/base.php (hunk 1 above)
Comment (pascalchevrel) [inline app/views/templates/base.php]: ok

**Comment by pascalchevrel** [comment]
If you switch locales temporarilly with the locale switcher widget, you can get results in your language.

**Comment by flodolo** [comment]
I'm not sure how intuitive it is, but I also don't think having 2 requests for each character makes sense (the alternative is adding an API call just for suggestions, which might be used also for a searchplugin).

Anyhow, it's an r+ from me :+1:

**Pull request merged**